File: src/contests/cg/puzzle/clojure/sum-of-divisors.clj
;; 🎮 CodinGame Puzzle - sum-of-divisors
;; https://www.codingame.com/training/medium/sum-of-divisors

(ns Solution (:gen-class))

;; d appears as a divisor in floor(n/d) of the numbers 1..n, so the total sum
;; of divisors is sum over d of d * floor(n/d).
(defn -main [& _]
  (let [n (Long/parseLong (.trim (read-line)))]
    (loop [d 1 total 0]
      (if (<= d n)
        (recur (inc d) (+ total (* d (quot n d))))
        (println total))))
  (flush))

File: src/contests/cg/puzzle/lua/sum-of-divisors.lua
-- 🎮 CodinGame Puzzle - sum-of-divisors
-- https://www.codingame.com/training/medium/sum-of-divisors

-- d appears as a divisor in floor(n/d) of the numbers 1..n, so the total sum
-- of divisors is sum over d of d * floor(n/d).
local n = tonumber(io.read())
local total = 0
for d = 1, n do
  total = total + d * (n // d)
end
print(total)
